### DISCUSSION TITLE
Webcams

### DISCUSSION THREAD

**Discussion by Mkorneli** [theyosh/TerrariumPI#974] [Q&A] [answered]
I've been seaching around and i'm not able to figure out the limitations in use of webcams.
Seems to be able to add one of each?
Rtsp only work as a static and not live cam. Even if i can get the live feed through VLC. Do I use the wrong codec?

So for now I have one USB cam connected and one rtsp stream.
Can i change the frequency of the update? For me it seems like it updates every 30 second. Is that correct?

So asking kindly about a bit more information regarding webcam use.

Eventually I want to have two(or more) live rtsp cameras in my setup.

Thanks

**Comment by theyosh** [comment, marked as answer]
Hi @Mkorneli 

it is a bit hard to explain easy :(.  But yes, the still image webcams will update every 30 seconds. Depending on the resource, the images has to be tiled and scaled to different sizes in order to get the zoom-able map view. And that takes a lot of time.

The RPI cam needs about 10 - 13 seconds of time for every update. So going faster, is almost impossible. When you add another full HD usb webcam to it, it will take at least another 10 seconds for the USB webcam. And than it can get difficult to update every 30 seconds. The webcams are processed in parallel, but still, the python code can only use 1 CPU :(. So more webcams will just take more time.

For live streaming, we are depending on the RPI GPU. As live video transcoding is not possible with the CPU. That will overload the PI. You can set the amount of GPU memory which should allow more live video transcoding on the GPU. But I have not tested that. As it is getting expensive buying Full HD camera's.  So I do not know if it is possible to have 2 or more live streaming webcams.

So I am not sure if it will work with 2 or more RTSP live webcams. That has not been tested. It could work if you set more memory to the GPU with raspi-config tool.

But again, I cannot guarantee that it will work.

**Comment by theyosh** [comment]
Any updates? Does the above explanation help?

**Reply by Mkorneli** [reply]
The explanation was good, but I know for a fact that there is solutions out there that gives an opportunity for "some what" live stream and several cameras. Even on RPI.
For now one is enough. 
Maybe something you could consider updating/expanding?
